src/ci/init_repo.sh

```diff
@@ -8,19 +8,17 @@ set -o errexit
 set -o pipefail
 set -o nounset
 
-ci_dir=$(cd $(dirname $0) && pwd)
+ci_dir=$(cd "$(dirname "$0")" && pwd)
 . "$ci_dir/shared.sh"
 
 REPO_DIR="$1"
 CACHE_DIR="$2"
 
-cache_src_dir="$CACHE_DIR/src"
-
-if [ ! -d "$REPO_DIR" -o ! -d "$REPO_DIR/.git" ]; then
+if [ ! -d "$REPO_DIR" ] || [ ! -d "$REPO_DIR/.git" ]; then
```

Comment:
If you're going to avoid inconsistent behavior in `test`, let's be consistent about it ;)
```suggestion
if ! { [ -d "$REPO_DIR" ] || [ -d "$REPO_DIR/.git" ] }; then
```

Reply:
Wouldn't this require the || to be changed to &&?

Reply:
Hm I implemented this and it caused a syntax error, so I'll leave it as it was for now

Reply:
You needed a semicolon before the `}`, I left it as a review comment.